server/services/gameTick.ts

```diff
@@ -155,12 +159,13 @@ export default class GameTickService extends EventEmitter {
         }
 
         /*
-            1. Move all carriers
-            2. Perform combat at stars that have enemy carriers in orbit
-            3. Industry creates new ships
-            4. Players conduct research
-            5. If its the last tick in the galactic cycle, all players earn money and experimentation is done.
-            6. Check to see if anyone has won the game.
+            1. Buy scheduled infrastructure
```

Comment:
Scheduling this prior to combat will result in players wasting credits they wouldn't on stars they may lose
Scheduling this prior to Industry, research & earning money gives players using this feature an advantage over those that don't

I think the fairest time to do this is after `5. If its the last tick in the galactic cycle, all players earn money and experimentation is done.`

Also most of the time this is only going to kick in after players earn money.

Reply:
I agree that this is the correct time. If you accidentaly buy infrastructure on stars that are lost during the tick, that would be no different from using bulk buy yourself during the tick before, and messing up with the ignored stars. This way, it is also impossible to spend your cash **during a tick** before you can login during that tick. Say I schedule a full econ buy before the cycle, and I login the moment before the tick happens, the money would all be spent already (if the order is changed). I think that is undesirable.